### PR TITLE
[popover2] feat: ContextMenu2 component

### LINKED ISSUE #3891: Unable to get ContextMenu working (HOC style)
#### Environment

- __Package version(s)__: @blueprintjs/core@3.22.2, 
- __Browser and OS versions__:  Version 1.1.20 Chromium: 79.0.3945.74 (Official Build) (64-bit), Linux 4.15.0-72-generic #81-Ubuntu SMP Tue Nov 26 12:20:02 UTC 2019

#### Question

Using the higher order component example from the docs, I am unable to get the ContextMenu to work. The error I am getting is

`TypeError: Class constructor RightClickMeWithContext cannot be invoked without 'new'`

Full error page:
https://pastebin.com/9N7Uusrb

I am using a `create-react-app` project, hence the HOC solution.

My component:
```
import React from "react";
import { ContextMenuTarget, Menu, MenuItem } from "@blueprintjs/core";

export const RightClickMe = ContextMenuTarget(
  class RightClickMeWithContext extends React.Component {
    render() {
      // root element must support `onContextMenu`
      return <div>TEST</div>;
    }

    renderContextMenu() {
      // return a single element, or nothing to use default browser behavior
      return (
        <Menu>
          <MenuItem onClick={this.handleSave} text="Save" />
          <MenuItem onClick={this.handleDelete} text="Delete" />
        </Menu>
      );
    }

    onContextMenuClose() {
      // Optional method called once the context menu is closed.
    }
  }
);
```

### LINKED ISSUE #4038: [core] isDarkTheme utility should be exported
along with the others in `src/common/utils/`

### PR DESCRIPTION
**Pull request by adidahiya**
#### Checklist

- [x] Includes tests
- [x] Update documentation

#### Changes proposed in this pull request:

New `<ContextMenu2>` component, works with hooks, no more potentially confusing decorators, simply works with Popover2 under the hood...

#### Reviewers should focus on:

API parity

#### Screenshot

image

### CODE REVIEW

File: packages/popover2/src/contextMenu2.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                placement="right-start"
+                rootBoundary="viewport"
+                renderTarget={renderTarget}
+                transitionDuration={TRANSITION_DURATION}
+            />
+            {children}
+        </div>
+    );
+};
+ContextMenu2.displayName = "Blueprint2.ContextMenu2";
```
Comment (adidahiya) [inline packages/popover2/src/contextMenu2.tsx]: extra `2` here

File: packages/popover2/src/_context-menu2.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Copyright 2021 Palantir Technologies, Inc. All rights reserved.
```
Comment (adidahiya) [inline packages/popover2/src/_context-menu2.scss]: need to remove this file

File: packages/docs-app/src/examples/core-examples/contextMenu2Example.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    <MenuItem icon="search-around" text="Search around..." />
+                    <MenuItem icon="search" text="Object viewer" />
+                    <MenuItem icon="graph-remove" text="Remove" />
+                    <MenuItem icon="group-objects" text="Group" />
+                    <MenuDivider />
+                    <MenuItem disabled={true} text="Clicked on node" />
+                </Menu>
+            }
+            onOpened={() => setIsOpen(true)}
+            onClosed={() => setIsOpen(false)}
```
Comment (adidahiya) [inline packages/docs-app/src/examples/core-examples/contextMenu2Example.tsx]: we should make this use case easier. `children` should support a render prop pattern which provides `isOpen` to the render callback

**Comment by stevestorey** [comment]
I've been using the `onNodeContextMenu` property of the `Tree` class https://blueprintjs.com/docs/#core/components/tree to show a context menu on tree nodes by calling `ContextMenu.show` - https://blueprintjs.com/docs/#core/components/context-menu.imperative-usage - but that doesn't seem to be an API supported in the new `ContextMenu2` component, and given I don't see a way to override the tree item renderer to be able to wrap the tree nodes in a `<ContextMenu2>` component, I'm not sure how best to migrate over.

How I can use the new ContextMenu2 to show context menus for tree nodes ?

thanks

**Comment by adidahiya** [comment]
@stevestorey can you open a new issue? or tack onto this related one: https://github.com/palantir/blueprint/issues/4537

**Pull request merged**